packages/blocks/src/note-block/keymap.ts

```diff
@@ -41,13 +41,37 @@ export function bindHotKey(blockElement: BlockElement) {
   });
   blockElement.bindHotKey({
     ArrowDown: () => {
+      let currentBlock: BlockElement | null = null;
       return root.std.command
         .pipe()
         .inline((_, next) => {
           reset();
           next();
         })
         .try(cmd => [
+          // text selection - case 1: move cursor down within the same block
+          cmd
+            .getTextSelection()
+            .inline<'currentSelectionPath'>((ctx, next) => {
```

Comment:
There seems to be a repetitive `inline` command in the logic of `try`, and I think we can add it before `try` to avoid repetition.

```ts
root.std.command
        .pipe()
        .inline((_, next) => {
          reset();
          next();
        })
        .inline<'currentSelectionPath'>((ctx, next) => {
              const textSelection = ctx.currentTextSelection;
              assertExists(textSelection);
              const end = textSelection.to ?? textSelection.from;
              next({ currentSelectionPath: end.path });
        })
        .try(...)
```

Reply:
@Flrande to move this part out of `try` block, I also have to move `getTextSelection` with it as it need `ctx.currentTextSelection`
```ts
root.std.command
        .pipe()
        .inline((_, next) => {
          reset();
          next();
        })
 -->   .getTextSelection()  
        .inline<'currentSelectionPath'>((ctx, next) => {
              const textSelection = ctx.currentTextSelection;
              assertExists(textSelection);
              const end = textSelection.to ?? textSelection.from;
              next({ currentSelectionPath: end.path });
        })
        .try(...)
```

This was failing in case selection is a block and does not reach the last condition of moving arrow for block selection present in the same try block.

I have now switched to a nested try block, where I am able to extract the repetitive code out of the inner try block.